pkg/dockercreds/k8sdockercreds/k8s_keychain.go

```diff
@@ -79,13 +79,10 @@ func (k *annotatedBasicAuthKeychain) Resolve(res authn.Resource) (authn.Authenti
 	sort.Slice(secrets, func(i, j int) bool { return secrets[i].Name < secrets[j].Name })
 
 	for _, s := range secrets {
-		if match(s, res) {
+		matcher := dockercreds.RegistryMatcher{Registry: s.Annotations[v1alpha1.DOCKERSecretAnnotationPrefix]}
```

Comment:
Do we still have the `v1.SecretTypeBasicAuth` check on docker basic auth secrets?

Reply:
Good catch..will fix and merge